File: dotnet_3/cs/rest/SCS/Imp/Cookie.cs
namespace SCS.Imp
{
    public class Cookie
    {
        public static string Subject(string name,  string val, string site)
        {
            name = name.ToLower();
            val = val.ToLower();
            site = site.ToLower();
            var result = 0;

            if (name == "userid") {
                if (val.Length > 6) {
                    if ("user".Equals(val[..4])) {
                        result = 1;
                    }
                }
            }
            else if (name == "session" && ("am".Equals(val) && "abc.com".Equals(site))) {
                result = 1;
            }
            else if (name == "session") {
                result = 2;
            }

            return "" + result;
        }
    }
}

File: jdk_8_maven/cs/graphql/graphql-scs/src/main/kotlin/org/graphqlscs/type/Cookie.kt
package org.graphqlscs.type

import org.springframework.stereotype.Component
import java.util.*
@Component
class Cookie {
    fun subject(name: String, `val`: String, site: String): String {
        var name = name
        var `val` = `val`
        var site = site
        name = name.lowercase(Locale.getDefault())
        `val` = `val`.lowercase(Locale.getDefault())
        site = site.lowercase(Locale.getDefault())
        var result = 0
        if ("userid" == name) {
            if (`val`.length > 6) {
                if ("user" == `val`.substring(0, 4)) {
                    result = 1
                }
            }
        } else if ("session" == name) {
            result = if ("am" == `val` && "abc.com" == site) {
                1
            } else {
                2
            }
        }
        return "" + result
    }
}